src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private double sideLength;
+
+    public Square(Color color, double sideLength) {
+        super(color, Math.pow(sideLength, 2));
```

Comment:
Creating a figure, don't pass expressions in the constructor. Instead, create a separate variable for the area calculation and pass it on for better code readability.